packages/react/widget/src/components/BaseWidget.tsx

```diff
@@ -0,0 +1,64 @@
+import { PropsWithChildren, useEffect, useRef } from 'react';
+import { usePortalWidgets } from '../hooks/use-portal-widgets';
+import { WidgetModule, WidgetProps } from '@equinor/fusion-framework-module-widget';
+import { Fusion } from '@equinor/fusion-framework';
+
+interface WidgetComponentProps<TProps extends WidgetProps> {
+    readonly props?: TProps;
+    readonly name: string;
+    readonly widgetVersion?: {
+        type: 'version' | 'tag';
+        value: string;
+    };
+    readonly fusion: Fusion<[WidgetModule]>;
+}
+
+export const BaseWidget = <TProps extends WidgetProps>({
+    name,
+    children,
+    props,
+    widgetVersion,
+    fusion,
+}: PropsWithChildren<WidgetComponentProps<TProps>>) => {
+    const ref = useRef<HTMLElement>(null);
+
+    const provider = fusion.modules.widget;
+    const { loading, error, widgetRef, widget } = usePortalWidgets(provider, () => ({
```

Comment:
see comment for this hook, think the functionality should just be here.

the `usePortalWidget` will create a new result for every render, since the callback is resolved on every call to the hook